main.c

```diff
@@ -139,25 +140,28 @@ static void print_p(const void * unused) { }
 static void hexdump(const void *unused, size_t unused2) { }
 #endif
 
-static void extend_pcr(struct tpm *tpm, void *data, u32 size, u32 pcr)
+static void extend_pcr(struct tpm *tpm, void *data, u32 size, u32 pcr, char *ev)
 {
-	if (tpm->family == TPM12) {
-		u8 hash[SHA1_DIGEST_SIZE];
+	u8 hash[SHA1_DIGEST_SIZE];
+	sha1sum(hash, data, size);
+	print("shasum calculated:\n");
+	hexdump(hash, SHA1_DIGEST_SIZE);
+	tpm_extend_pcr(tpm, pcr, TPM_HASH_ALG_SHA1, hash);
 
-		sha1sum(hash, data, size);
-		print("shasum calculated:\n");
-		hexdump(hash, SHA1_DIGEST_SIZE);
-		tpm_extend_pcr(tpm, pcr, TPM_ALG_SHA1, hash);
-		print("PCR extended\n");
+	if (tpm->family == TPM12) {
+		log_event_tpm12(pcr, hash, ev);
 	} else if (tpm->family == TPM20) {
 		u8 sha256_hash[SHA256_DIGEST_SIZE];
 
 		sha256sum(sha256_hash, data, size);
 		print("shasum calculated:\n");
 		hexdump(sha256_hash, SHA256_DIGEST_SIZE);
-		tpm_extend_pcr(tpm, pcr, TPM_ALG_SHA256, &sha256_hash[0]);
-		print("PCR extended\n");
+		tpm_extend_pcr(tpm, pcr, TPM_HASH_ALG_SHA256, &sha256_hash[0]);
+
+		log_event_tpm20(pcr, hash, sha256_hash, ev);
```

Comment:
It looks like SHA1 is extended for both 1.2 and 2.0 unconditionally. The event sent to the event log should also contain both hashes to be an accurate record of what went into the TPM. This is also consistent with the way the ACM and the SL code in the kernel currently handle things.

Reply:
And that is what this function does: `hash` is SHA1 and `sha256_hash` is SHA256. As both algorithms are required by TPM spec there is no reason to not do so, as SHA1 functions are compiled in anyway.

Reply:
Oh I see. Sorry I missed that. Perfect, thanks.